server/asset/btc/btc_test.go

```diff
@@ -1226,6 +1226,39 @@ func TestReorg(t *testing.T) {
 			test(a, b)
 		}
 	}
+
+	// Create a transaction at the tip, then orphan the block and move the
+	// transaction to mempool.
+	reset()
+	setChain(chainA)
+	tipHeight := btc.blockCache.tipHeight()
+	txHash := randomHash()
+	tip := btc.blockCache.mainchain[uint32(tipHeight)]
+	msg := testMakeMsgTx(false)
+	testAddBlockVerbose(&tip.hash, nil, 1, uint32(tipHeight))
+	testAddTxOut(msg.tx, 0, txHash, &tip.hash, int64(tipHeight), 1)
+	utxo, err := btc.utxo(txHash, msg.vout, nil)
+	if err != nil {
+		t.Fatalf("utxo error: %v", err)
+	}
+	confs, err := utxo.Confirmations()
+	if err != nil {
+		t.Fatalf("Confirmations error: %v", err)
+	}
+	if confs != 1 {
+		t.Fatalf("wrong number of confirmations. expected 1, got %d", confs)
+	}
+
+	// Orphan the block and move the transaction to mempool.
```

Comment:
Would be good to test the transaction being found in a lower height block after reorg.  The same `!found` condition in in `(*TXIO).confirmations`, but with a positive confs value